# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.h
# Commit: b99b71e755e4
# Full Hash: b99b71e755e470a5dd152a5d6313f3002b2b9ba9
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-12-18 09:43:08
# Description:
#   Bug 1744081 - Move VideoConduit::Shutdown asyncness ownership into VideoConduit. r=bryce,ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D133933
# ==============================================================================

diff -r f69c775a1024 -r b99b71e755e4 dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.h
--- a/dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.h	Fri Dec 17 20:29:15 2021 +0000
+++ b/dom/media/webrtc/libwebrtcglue/WebrtcVideoCodecFactory.h	Fri Dec 17 20:34:18 2021 +0000
@@ -12,6 +12,7 @@
 
 namespace mozilla {
 class GmpPluginNotifierInterface {
+  virtual void DisconnectAll() = 0;
   virtual MediaEventSource<uint64_t>& CreatedGmpPluginEvent() = 0;
   virtual MediaEventSource<uint64_t>& ReleasedGmpPluginEvent() = 0;
 };
@@ -23,18 +24,19 @@
         mCreatedGmpPluginEvent(mOwningThread),
         mReleasedGmpPluginEvent(mOwningThread) {}
 
-  ~GmpPluginNotifier() {
+  ~GmpPluginNotifier() = default;
+
+  void DisconnectAll() override {
+    MOZ_ASSERT(mOwningThread->IsOnCurrentThread());
     mCreatedGmpPluginEvent.DisconnectAll();
     mReleasedGmpPluginEvent.DisconnectAll();
   }
 
   MediaEventSource<uint64_t>& CreatedGmpPluginEvent() override {
-    MOZ_ASSERT(mOwningThread->IsOnCurrentThread());
     return mCreatedGmpPluginEvent;
   }
 
   MediaEventSource<uint64_t>& ReleasedGmpPluginEvent() override {
-    MOZ_ASSERT(mOwningThread->IsOnCurrentThread());
     return mReleasedGmpPluginEvent;
   }
 
@@ -102,6 +104,8 @@
   std::unique_ptr<webrtc::VideoEncoder> CreateVideoEncoder(
       const webrtc::SdpVideoFormat& aFormat) override;
 
+  void DisconnectAll() override { mInternalFactory->DisconnectAll(); }
+
   MediaEventSource<uint64_t>& CreatedGmpPluginEvent() override {
     return mInternalFactory->CreatedGmpPluginEvent();
   }